int8 type
int16 rate
int32 accuracy
int8 inUse
